docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,236 @@
+the current set of load balancers used in the cloud rely on a separate cookie for implementing sticky sessions (e.g. HAProxy),
+and thus the default WildFly configuration lacks the ability to allow WildFly to guide requests for a given session to the server that can most efficiently handle it.
+To resolve this, we need to configure the Undertow subsystem to use a separate cookie to send affinity information.
+Now that the configured cache is using the distribution mode, the load balancer will need to be made aware of the affinity.
+As the topology of the cluster changes, the affinity will need to be updated.
+
+The following `undertow.cli` configuration script will reconfigure the `undertow` subsystem to use a separate cookie named `INGRESSCOOKIE` to store the affinity information:
+
+[source,cli,title=undertow.cli]
+----
+/subsystem=undertow/servlet-container=default/setting=affinity-cookie:add(name=INGRESSCOOKIE)
+----
+
+=== Installing HAProxy Ingress Controller
+
+The HAProxy Ingress Controller can be installed using https://helm.sh/[Helm]. First add the required repository:
+
+[source,shell]
+----
+$ helm repo add haproxy-ingress https://haproxy-ingress.github.io/charts
+"haproxy-ingress" has been added to your repositories
+----
+
+Now, the Ingress Controller can be installed to the Kubernetes cluster.
+
+[source,shell]
+----
+$ helm install haproxy-ingress haproxy-ingress/haproxy-ingress --create-namespace --namespace=ingress-controller
+NAME: haproxy-ingress
+LAST DEPLOYED: Tue Nov 14 16:18:10 2023
+NAMESPACE: ingress-controller
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+NOTES:
+HAProxy Ingress has been installed!
+
+HAProxy is exposed as a `LoadBalancer` type service.
+It may take a few minutes for the LoadBalancer IP to be available.
+You can watch the status by running:
+
+    kubectl --namespace ingress-controller get services haproxy-ingress -o wide -w
+----
+
+Once the controller is available in the Kubernetes cluster, we can configure an Ingress object.
+
+=== Configuring an Ingress
+
+The following Ingress configuration can be used for optimal handling of session affinity.
+Note that the name of the ingress, hosts, and the service name needs to be updated to actual values.
+
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+  namespace: default
+  annotations:
+    haproxy-ingress.github.io/affinity: cookie # <1>
+    haproxy-ingress.github.io/backend-server-naming: pod # <2>
+    haproxy-ingress.github.io/session-cookie-dynamic: "false" # <3>
+    haproxy-ingress.github.io/session-cookie-keywords: preserve # <4>
+    haproxy-ingress.github.io/session-cookie-name: INGRESSCOOKIE # <5>
+    haproxy-ingress.github.io/session-cookie-preserve: "true" # <6>
+    haproxy-ingress.github.io/session-cookie-strategy: insert # <7>
+    haproxy-ingress.github.io/session-cookie-value-strategy: server-name # <8>
```

Comment:
We also want to override `haproxy-ingress.github.io/balance-algorithm` to use `source` or `hdr(Cookie)` (if that works) so that failover targets are chosen deterministically.

Reply:
@pferraro Good idea - opened https://issues.redhat.com/browse/WFLY-19252 to investigate.